.github/container/jax_nsys/Analysis.ipynb

```diff
@@ -845,9 +832,17 @@
     "    squeeze=False,\n",
     ")\n",
     "for n_module, module_id in enumerate(module_ids_with_traces):\n",
-    "    proto = xla_module_metadata(module_id).proto().buffer_assignment\n",
-    "    sizes_by_logical_id = {buffer.id: buffer.size for buffer in proto.logical_buffers}\n",
-    "    for n_trace, trace in enumerate(proto.heap_simulator_traces):\n",
+    "    protos = xla_module_metadata(module_id, policy=\"all\")\n",
+    "    sizes_by_logical_id = protos.unique_result(\n",
+    "        lambda proto: {\n",
+    "            buffer.id: buffer.size\n",
+    "            for buffer in proto.proto().buffer_assignment.logical_buffers\n",
+    "        }\n",
+    "    )\n",
+    "    traces = protos.unique_result(\n",
```

Comment:
Yeah, I think that's a good point. I think the compute/communication overlap is another example where the current implementation needs some improvement to scale to larger profile datasets.

For the protobuf metadata specifically, if it's actually identical across ranks then `nsys-jax-combine` will de-duplicate, but at the moment that is all-or-nothing and I think at the moment there is never a bitwise match anyway. However, it might only need a small change either to XLA (to remove the source of mismatches) or to the `nsys-jax-combine` de-duplication logic (to ignore some known-harmless cases) to make it so that `unique_result` is operating over `O(1)` instead of `O(R)` values.